frontend/src/App.jsx

```diff
@@ -1,35 +1,47 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import './App.css';
+import './components/listoftodo/ListOfTodo'
+import firebase from 'firebase/compat/app';
+import 'firebase/compat/auth';
+import { useState, useEffect } from 'react';
+import ListOfTodo from './components/listoftodo/ListOfTodo';
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [auth, setAuth] = useState(false || window.localStorage.getItem('auth')=='true');
+  const [token, setToken] = useState('');
 
+
+  useEffect(() => {
+    firebase.auth().onAuthStateChanged((userCred)=>{
+      if(userCred){
+        setAuth(true);
+        window.localStorage.setItem('auth','true');
+        userCred.getIdToken().then((token)=>{
+          setToken(token);
+        });
+      }
+    })
+  }, []);
+
+  const loginWithGoogle = () => {
```

Comment:
So I see you have a button for login - you probably need one for creating an account too, right?

Or does FireBase's UI handle both the creating account and logging in?

If so, what APIs does firebase provide so you can handle "create account" and "log in" separately?